testing/app.go

```diff
@@ -58,16 +58,20 @@ func SetupTestingApp() (TestingApp, map[string]json.RawMessage) {
 // that also act as delegators. For simplicity, each validator is bonded with a delegation
 // of one consensus engine unit (10^6) in the default token of the simapp from first genesis
 // account. A Nop logger is set in SimApp.
-func SetupWithGenesisValSet(t *testing.T, valSet *tmtypes.ValidatorSet, genAccs []authtypes.GenesisAccount, chainID string, balances ...banktypes.Balance) TestingApp {
+func SetupWithGenesisValSet(t *testing.T, valSet *tmtypes.ValidatorSet, genAccs []authtypes.GenesisAccount, chainID string, powerReduction sdk.Int, balances ...banktypes.Balance) TestingApp {
 	app, genesisState := DefaultTestingAppInit()
+
 	// set genesis accounts
-	authGenesis := authtypes.NewGenesisState(authtypes.DefaultParams(), genAccs)
+	var authGenesis *authtypes.GenesisState
+	app.AppCodec().MustUnmarshalJSON(genesisState[authtypes.ModuleName], authGenesis)
+
+	authGenesis = authtypes.NewGenesisState(authGenesis.Params, genAccs)
 	genesisState[authtypes.ModuleName] = app.AppCodec().MustMarshalJSON(authGenesis)
 
 	validators := make([]stakingtypes.Validator, 0, len(valSet.Validators))
 	delegations := make([]stakingtypes.Delegation, 0, len(valSet.Validators))
 
-	bondAmt := sdk.NewInt(1000000)
+	bondAmt := sdk.TokensFromConsensusPower(1, powerReduction)
```

Comment:
Would it work for you if we just made this a global variable in the tests that you could change? Seems odd to add this as a function argument.

Reply:
I could use the default consensus power (global from the SDK), but then it wouldn't support testing between chains with different power values

Reply:
We want tests that involve powers such as

ChainA: 10e6
ChainB: 10e18

Reply:
When chains change the power reduction, do they just change the global in the SDK, i.e. there is no on-chain parameter? I'm wondering if there's a way we can allow this information to be specified in `DefaultTestingAppInit`

Reply:
> When chains change the power reduction, do they just change the global in the SDK, i.e. there is no on-chain parameter?

yes, there's no chain parameter. If we just change the parameters, the 2 chains from the tests will share the same power. That's why I'm adding it as a field so that we can tests txs between chains with different power values